cmd/network/verification.go

```diff
@@ -290,6 +290,11 @@ func (e *egressVerification) getSubnetId(ctx context.Context) (string, error) {
 		return e.cluster.AWS().SubnetIDs()[0], nil
 	}
 
+	if e.cluster.Hypershift().Enabled() {
+		e.log.Info(ctx, "detected BYOVPC Hypershift cluster, using first subnet from OCM: %s", e.cluster.AWS().SubnetIDs()[0])
+		return e.cluster.AWS().SubnetIDs()[0], nil
```

Comment:
HyperShift clusters are always BYOVPC, but they don't have to be PrivateLink, so I don't think we can just grab the first subnet from OCM (since the non-PrivateLink BYOVPC HyperShift clusters will also have public subnets stored in OCM)

Reply:
good point! I've updated here to a form where the subnetID is queried from worker nodes of the hcp cluster